src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/volume/v2/OpenStackVolumePluginTest.java

```diff
@@ -1,201 +1,524 @@
+                OpenStackStateMapper.IN_USE_STATUS };
+
+        for (String cloudState : cloudStates) {
+            // exercise
+            boolean status = this.plugin.isReady(cloudState);
+
+            // verify
+            Assert.assertFalse(status);
+        }
+    }
+    
+    // test case: When calling the hasFailed method with any cloud states errors,
+    // this means that the state of the volume is FAILED and it must return true.
+    @Test
+    public void testHasFailedSuccessful() {
+        // set up
+        String[] cloudStates = { OpenStackStateMapper.ERROR_BACKING_UP_STATUS,
+                OpenStackStateMapper.ERROR_DELETING_STATUS, OpenStackStateMapper.ERROR_EXTENDING_STATUS,
+                OpenStackStateMapper.ERROR_RESTORING_STATUS, OpenStackStateMapper.ERROR_STATUS };
+
+        for (String cloudState : cloudStates) {
+            // exercise
+            boolean status = this.plugin.hasFailed(cloudState);
+
+            // verify
+            Assert.assertTrue(status);
+        }
     }
+    
+    // test case: When calling the hasFailed method with the cloud states different
+    // than any error, this means that the state of the volume is not FAILED and it
+    // must return false.
+    @Test
+    public void testhasFailedUnsuccessful() {
+        // set up
+        String[] cloudStates = { OpenStackStateMapper.CREATING_STATUS, OpenStackStateMapper.AVAILABLE_STATUS,
+                OpenStackStateMapper.IN_USE_STATUS };
+
+        for (String cloudState : cloudStates) {
+            // exercise
+            boolean status = this.plugin.hasFailed(cloudState);
 
-    // test case: Check if the request in requestInstance() is executed properly with the right parameters even when
-    // there is volume extra requirements.
+            // verify
+            Assert.assertFalse(status);
+        }
+    }
+    
+    // test case: When invoking the requestInstance method with a valid volume
+    // request and a cloud user, it must verify that the call was successful.
     @Test
-    public void testRequestInstanceWithRequirements() throws FogbowException, HttpResponseException {
+    public void testRequestInstance() throws FogbowException {
         // set up
-        VolumeOrder volumeOrder = new VolumeOrder(null, null, "fake-name", 2);
-        Map<String, String> requirements = new HashMap<>();
-        requirements.put("fake-capabilities", "fake-value");
-        volumeOrder.setRequirements(requirements);
+        VolumeOrder order = this.testUtils.createLocalVolumeOrder();
+        OpenStackV3User cloudUser = createOpenStackUser();
 
-        Mockito.doReturn(FAKE_TYPES_JSON).when(this.client).doGetRequest(
-                Mockito.anyString(), Mockito.any(OpenStackV3User.class));
-        Mockito.doReturn(FAKE_VOLUME_JSON).when(this.client).doPostRequest(
-                Mockito.anyString(), Mockito.any(), Mockito.any(OpenStackV3User.class));
+        PowerMockito.mockStatic(OpenStackCloudUtils.class);
```

Comment:
Is this mock necessary? It seem's it is mockinf the CloudUtils to have this exact same behaviour. The same apply for below ocurrencies.

Reply:
The intention is to verify that this method was actually called. I'll change the mock method to thenCallRealMethod;